rpg.py

```diff
@@ -120,47 +129,45 @@ def act(self, players):
 
         return players
 
-    def move_input(self):
+    def move_input(self, players: List[Character]):
         move = None
         while move is None:
-            m_input = input('>')
-            move = m_input
+            move_input = input('>')
+            move = move_input
             if move == 'f':
-                pass
-            elif move == 'i':
+                break
+            if move == 'i':
                 self.use_item()
-            elif move == 's':
+            if move == 's':
                 self.get_stats()
-                self.move_input()
             else:
-                self.move_input()
-            return move
+                return self.move_input(players)
 
     def use_item(self):
-        if potion.name in inventory:
+        if potion in inventory:
             print(f'would you like to use {potion.name}?')
             print('[y]es or [n] no')
-            item_input = input('>')
-            if item_input == 'y':
-                if self.hp < self.max_hp:
-                    self.hp += potion.hp_restore
-                    inventory.remove(potion.name)
-                else:
-                    print('It won\'t help')
-                    self.move_input()
-            if item_input == 'n':
-                self.move_input()
+            item_use = None
+            while item_use is None:
```

Comment:
What's the point of `item_use` if you never change it? You can just do `while True:`

I would also advise against doing something like that anyways. It's a bad coding pattern to using unconditional looping then a `break`. It's hard to follow and beats the purpose of having a looping condition.